### PR TITLE
feat: iavl async pruning

### PR DESCRIPTION
**Pull request by cool-develope**
## Description

- introduce new interfaces `SetCommitting` and `UnsetCommitting` to block the pruning on the iavl side 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

## Summary by CodeRabbit

- **New Features**
	- Implemented asynchronous pruning to enhance performance.
	- Added functionality to mark store states as committing or not, improving transaction handling and state management.

- **Tests**
	- Introduced delays in test scenarios to better simulate consensus processes and ensure accurate pruning behavior during commits.

### CODE REVIEW

**Comment by nicolaslara** [comment]
Thanks for the submission. The upstream IAVL change seems to be on a branch though. We will want to wait for that to be merged to mainline and tagged before merging this. Is there a PR where that change is proposed?

**Review by czarcas7ic — APPROVED** [review]
This feat is going to be merged upstream. We tested against Osmosis v23 mainnet and it seems to have fixed the issue of nodes halting for pruning

**Pull request merged**
